class.jetpack.php

```diff
@@ -3018,7 +2891,7 @@ function admin_init() {
 		// If the plugin is not connected, display a connect message.
 		if (
 			// the plugin was auto-activated and needs its candy
-			Jetpack_Options::get_option( 'do_activate' )
+			Jetpack_Options::get_option_and_ensure_autoload( 'do_activate', '0' )
```

Comment:
`Jetpack_Options::get_option` prefixes the option passed with `jetpack_`.

I think we should incorporate the logic of `Jetpack_Options::get_option_and_ensure_autoload` to `Jetpack_Options::get_option` and avoid having a third way of getting options, what do you think?